hvantk/algorithms/qtlcascade/finemap.py

```
@@ -0,0 +1,289 @@
+
+
+def finemap_available() -> tuple[bool, list[str]]:
+    """Return ``(ok, missing)`` — whether R+packages and bcftools are present."""
+    missing: list[str] = []
+    if shutil.which("bcftools") is None:
+        missing.append("bcftools")
+    if shutil.which("Rscript") is None:
+        missing.append("Rscript (R)")
+        return False, missing
+    try:
+        r = subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)